docs/developer/continuous-deployment/continuous_deployment.md

```diff
@@ -27,11 +27,52 @@ You will need to provide the following:
 - Two service principals (instructions to create them below)
 - A fork of the [MVD repository](https://github.com/eclipse-dataspaceconnector/MinimumViableDataspace) and permissions to configure GitHub secrets
 
-## Create a Service Identity for GitHub Actions
+## Option 1: Automatically set up Azure AD using terraform 
+
+This is the easiest way to set up the Service Principals in Azure Active Directory, but it requires that the following tools are installed:
+- Azure CLI: required by Terraform
+- Terraform: used to create App Registrations and Service Principals in Azure AD
+- (optional) Github CLI: if secrets should be stored using the provided shell script
+
+### Log in to Azure CLI
+
+Install and login to the [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli). On a shell execute `az login` - your user must have "Application Administrator" rights in the Azure tenant.
+
+### Initialize terraform
+
+Assuming terraform is installed, simply `cd` to `resources/setup_azure_ad` (located in the repository root) and execute
+```shell
+terraform init
+```
+
+### Customize `variables.tf`
+
+Please change variables in [variables.tf](../../../resources/setup_azure_ad/variables.tf) to reflect your local setup. Note that `tenant_id` and `gihub_repo` **must** be changed.
+Note also that the terraform state file (`terraform.state`) will be stored locally in the same directory. This will contain sensitive information - **do not check it in**!
```

Comment:
Why not adding the state file to .gitignore?

Reply:
it is. there is one in `resources/setup_azure_ad`

Reply:
I just saw it is already there, is it really needed to mention that it must not be checked in?

Reply:
better to be safe than sorry